docs/middleware.rst

```diff
@@ -113,3 +113,31 @@ Middleware
 
       The deserialized JSON representation of the event that triggered the request if it exists, or ``None``.
       This header is set by the `event-header htmx extension <https://htmx.org/extensions/event-header/>`__, and contains details of the DOM event that triggered the request.
+
+
+.. function:: redirect_middleware(get_response)
+
+   Middleware to handle redirects for htmx requests for login required views.
+
+   :param get_response: A function that takes a request and returns a response.
+   :type get_response: callable
+   :returns: A middleware function that takes a request and returns a response.
+
+   The middleware function checks if the request is an htmx request and if the response status code is 302 (redirection). If so, it parses the redirection location. If the redirection location is the login URL, it changes the redirection URL to point to the login redirect URL and changes the response status code to 204 (No Content). The modified response is then returned.
+
+   .. code-block:: python
+
+     def redirect_middleware(get_response):
```

Comment:
copy-pasting the source code into the docs isn't necessary. they would drift over time.

Reply:
I think you're right, it does seem to be project-specific what should be done in cases like these. Thanks for taking the time to review!